charts/camunda-bpm-platform/templates/deployment.yaml

```diff
@@ -93,12 +93,28 @@ spec:
           volumeMounts:
```

Comment:
@kostkaf this part should be adjusted because now extra extraVolumeMounts only work if the database is H2.

Reply:
@kostkaf ahaa I see :+1: 

But in that case, the H2 and extra volumes are exclusive mutual, which is not necessary here.
We need to have a nested if condition for both of them to make them work together.

Reply:
It shouldn't be exclusive mutual because of the [else condition in line 99](https://github.com/camunda-community-hub/camunda-helm/blob/98e143caface9ee4cc0a46fc8ea1dbd1a4a744dc/charts/camunda-bpm-platform/templates/deployment.yaml#L99).

The "volumeMount" in line 101 is not rendered if H2 is used.

Reply:
No wonder I didn't see it :grinning:
TBH That's confusing (and redundant).

In all cases, I think we can move forward with the current proposal and I have an idea to make it more clear using the helper templates.

Reply:
@kostkaf @FabiusE FYI, here is the simplified version where no need to duplicate the `volumes` and `volumeMounts`.
https://github.com/camunda-community-hub/camunda-helm/blob/eb98e735d5b1c2abc7817bdd7af1741d5bc91399/charts/camunda-bpm-platform/templates/deployment.yaml#L94